### PR TITLE
mapcss parsing: string with int comparison triggers RuleAbort

### PR DESCRIPTION
**Pull request by Famlam**
Have this as an example (minimal) mapcss rule:
```css
*[x!=0] {
  assertMatch: "node x=\"0.5 osmose\"";
  throwWarning: "x";
}
```

Compile the python code and run the test. The match fails due to `modules.mapcss_lib.RuleAbort` being raised by `to_n` - called by `__ne__` - as it tries to convert the nonconvertible string `0.5 osmose` to a float:
https://github.com/osm-fr/osmose-backend/blob/fe158f2c45eac2d0a210cd22a8d0f0d9af68087e/mapcss/mapcss_lib.py#L158-L165
(The error is caught by the `try`/`except` in the mapcss2osmose-generated .py file)

For comparison: JOSM only uses numerical comparisons if the types match: https://josm.openstreetmap.de/browser/josm/trunk/src/org/openstreetmap/josm/gui/mappaint/mapcss/Functions.java#L722

Unfortunately there was no assertMatch in any mapcss file that revealed this behaviour. I've added one to the Dutch rules now (as this one contains a rule that did trigger the error, thereby causing false negatives)

p.s.: please note that the mapcss update was necessary as there's also a difference in how JOSM and Osmose treat `[/regex/!~/regex/]` rules, and fixing this bug revealed that difference. Since this is undocumented behaviour, I decided to just modify the Dutch mapcss so that both give the same result. Unfortunately I made a little mistake there, hence the double mapcss update.

### CODE REVIEW

**Comment by jocelynj** [comment]
It would be nice to add unit tests to mapcss_lib.py to catch this kind of issue. Is is possible?

**Comment by Famlam** [comment]
I kind-of figured that having an assert in "my" Dutch rules now also counts as a unit test, but yeah, maybe good to make a fake mapcss file with difficult cases (as the Dutch mapcss file may change in the future). Will do so once this patch lands, if that's fine with you?

**Comment by frodrigo** [comment]
> hence the double mapcss update.

Note, you can edit history in local and push force the branch to change the history.

**Comment by frodrigo** [comment]
> I kind-of figured that having an assert in "my" Dutch rules now also counts as a unit test, but yeah, maybe good to make a fake mapcss file with difficult cases (as the Dutch mapcss file may change in the future). Will do so once this patch lands, if that's fine with you?

May be just add test to `mapcss_lib.py`, for now, there is no test on it :-/.

**Comment by Famlam** [comment]
> > hence the double mapcss update.
> 
> Note, you can edit history in local and push force the branch to change the history.

Will try, one second

**Comment by Famlam** [comment]
> > I kind-of figured that having an assert in "my" Dutch rules now also counts as a unit test, but yeah, maybe good to make a fake mapcss file with difficult cases (as the Dutch mapcss file may change in the future). Will do so once this patch lands, if that's fine with you?
> 
> May be just add test to `mapcss_lib.py`, for now, there is no test on it :-/.

Ok. Probably on Friday or Saturday though, bit busy coming days so then this PR has to wait, sorry 0:)
Note that I also added `assertMatch:"way highway=residential traffic_sign=\"NL:C02\" oneway:bicycle=no oneway=no"` to the Dutch rules which would've triggered this issue prior to the patch, due to `oneway=no` (from assertMatch) vs `[oneway!=-1]` (from the rules), so there actually is a test on it now.

Do you mind if I create a generic test for mapcss rules? Even though some tests are "off-topic" to this PR? Because if we start testing, I'd prefer to also add tests for other "previous" issues of the mapcss parsing.

**Comment by frodrigo** [comment]
> Do you mind if I create a generic test for mapcss rules? Even though some tests are "off-topic" to this PR? Because if we start testing, I'd prefer to also add tests for other "previous" issues of the mapcss parsing.

Sure, tests are very welcome!

**Comment by Famlam** [comment]
Had some unexpected free time, so I implemented some tests :)
Figured this way made more sense than adding it to mapcss_lib, as this way the entire process is being tested, from compiling to parsing the rule to the specific functions involved. (And with few exceptions, almost everything can be tested with the asserts of mapcss). 
(Found the bug #1742 this way too; already added it to the file, but commented out for now until its fixed)

File: plugins/tests/test_mapcss_parsing_evalutation.validator.mapcss [hunk 1]
```diff
@@ -0,0 +1,162 @@
+/* Disclaimer: 
+This document only serves to contain test cases to evaluate
+the parsing and evaluation of mapcss files, and is therefore
+not supposed to be used in JOSM for validation of OSM data */
+
+
+node[x=0] {
+  throwWarning: "test #1740";
+	assertMatch: "node x=0";
```
Comment (frodrigo) [inline plugins/tests/test_mapcss_parsing_evalutation.validator.mapcss:9]: Can you fix the indentation of this file please.

Comment (Famlam) [reply plugins/tests/test_mapcss_parsing_evalutation.validator.mapcss]: Oops, didn't notice it. Fixed now

**Comment by frodrigo** [comment]
Nice. Thank you to do it.

**Pull request merged**